src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTask.cs

```diff
@@ -123,7 +124,36 @@ public override bool Execute()
                 if (string.IsNullOrEmpty(request.NuspecFile))
                 {
                     packageBuilder = logic.GetPackageBuilder(request);
+                    string noWarnProperties = null;
+
+                    if (packageBuilder?.Properties?.TryGetValue("NoWarn", out noWarnProperties) == true
+                        && !string.IsNullOrWhiteSpace(noWarnProperties))
+                    {
+                        PackageSpecificWarningProperties packageSpecificWarningProperties = new();
+
+                        foreach (string packageNoWarnProperty in noWarnProperties.Split(new string[] { Environment.NewLine },
+                            StringSplitOptions.RemoveEmptyEntries))
+                        {
+                            string[] packageNoWarnPropertyPars = packageNoWarnProperty.Split(';');
```

Comment:
We have this `MSBuildStringUtility` class that has some tried and true split logic, do we have a utility class that does this kind of parsing (semicolon delimited, key/value pairs)?

https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Common/MsBuildStringUtility.cs